src/main/kotlin/net/maizegenetics/phgv2/cli/CreateFastaFromHvcf.kt

```diff
@@ -66,33 +69,55 @@ class CreateFastaFromHvcf : CliktCommand( help = "Create a FASTA file from a h.v
         .default("")
 
 
+    // Pre-compile the Regex pattern - used when creating the output fasta file names
+    val HVCF_PATTERN = Regex("""(\.hvcf|\.h\.vcf|\.hvcf\.gz|\.h\.vcf\.gz)$""")
+
     /**
      * Function to build the Fasta file from the HVCF and the agc record.
      * Right now it does not support pulling from TileDB, but will in the future.
      */
-    fun buildFastaFromHVCF(dbPath: String, outputFile: String, fastaType:String, hvcfDir: String ,hvcfFile : String, condaEnvPrefix:String) {
-        if(hvcfDir != "") {
-            //Loop through the directory and figure out which files are hvcf files
-            // the gvcf and hvcf files may be in the same folder, so verify specific extension
-            val hvcfFiles = File(hvcfDir).listFiles { file -> file.extension == "h.vcf" || file.name.endsWith("h.vcf.gz") || file.name.endsWith("hvcf") || file.name.endsWith("hvcf.gz")}
-            //Loop through each file and run the buildFastaFromHVCF function
-            BufferedWriter(FileWriter(outputFile)).use { output ->
-                writeSequences(output,hvcfFiles.flatMap { processSingleHVCF(VCFFileReader(it,false), dbPath,condaEnvPrefix) }
-                    .associate { it.id to it }
-                    .values.toList(), fastaType)
+    fun buildFastaFromHVCF(dbPath: String, outputDir: String, fastaType:String, hvcfDir: String ,hvcfFile : String, condaEnvPrefix:String) {
+
+        if (hvcfDir != "") {
+            // Loop through the directory and figure out which files are hvcf files
+            // The gvcf and hvcf files may be in the same folder, so verify specific extension
+            val hvcfFiles = File(hvcfDir).listFiles { file ->
+                file.name.endsWith("h.vcf") || file.name.endsWith("h.vcf.gz") || file.name.endsWith("hvcf") ||
+                        file.name.endsWith("hvcf.gz")
+            }
+
+            // Loop through each file and run the processSingleHVCF function
+            hvcfFiles?.forEach { hvcfFile ->
+                val tempFileName = hvcfFile.name.replace(HVCF_PATTERN, ".fa")
```

Comment:
Can we get rid of tempFileName?  From a quick glance its only used to build the outputFileName.

Reply:
yes, updated